### PR TITLE
Activate WELPI for use in PYACTION

### PR DESCRIPTION
**Pull request by lisajulia**
Test is in opm-tests

### CODE REVIEW

**Comment by lisajulia** [comment]
jenkins build this opm-simulators=5463 opm-tests=1190 please

**Comment by akva2** [comment]
jenkins is not ready, please don't.

**Comment by lisajulia** [comment]
> jenkins is not ready, please don't.

Ah ok :) - thanks!

**Comment by lisajulia** [comment]
jenkins build this opm-simulators=5463 opm-tests=1190 please

**Comment by blattms** [comment]
mmh. Somehow this looks like the timestepping did change for the existing regression tests.

**Comment by lisajulia** [comment]
> mmh. Somehow this looks like the timestepping did change for the existing regression test mpi.compareParallelSim_flow+ACTIONX_WELPI
> 
> ```
> Comparing summary files 
> 
> Checking 271  vectors  ... Program threw an exception: [/var/lib/jenkins/post-builder/workspace/opm-common-PR-builder/test_util/EclRegressionTest.cpp:1064] 
> Keyword DAY summary vector of different length (76 != 74)
> ```

This is a test that I actually added, yet this should still pass....

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 1]
```diff
@@ -1439,6 +1440,11 @@ File {} line {}.)", pattern, location.keyword, location.filename, location.linen
         ScheduleGrid grid(this->completed_cells);
         SimulatorUpdate sim_update;
         std::unordered_map<std::string, double> target_wellpi;
+        // Get the well production indices from the simulation up until now
+        // and use in the calculation for the overall production indices.
+        if (this->wellPIPointer) {
+            target_wellpi = *(this->wellPIPointer);
+        }
```
Comment (blattms) [inline opm/input/eclipse/Schedule/Schedule.cpp:1447]: Might this be a change of behavior outside of PYACTION?

File: opm/input/eclipse/Schedule/Schedule.cpp [hunk 2]
```diff
@@ -1640,6 +1646,8 @@ File {} line {}.)", pattern, location.keyword, location.filename, location.linen
 
 
     SimulatorUpdate Schedule::runPyAction(std::size_t reportStep, const Action::PyAction& pyaction, Action::State& action_state, EclipseState& ecl_state, SummaryState& summary_state) {
+        // Set welpi_action_mode to true, this is necessary for the keyword WELPI
```
Comment (blattms) [inline opm/input/eclipse/Schedule/Schedule.cpp:1649]: Maybe you should explain why?

**Comment by lisajulia** [comment]
> > mmh. Somehow this looks like the timestepping did change for the existing regression test mpi.compareParallelSim_flow+ACTIONX_WELPI
> > ```
> > Comparing summary files 
> > 
> > Checking 271  vectors  ... Program threw an exception: [/var/lib/jenkins/post-builder/workspace/opm-common-PR-builder/test_util/EclRegressionTest.cpp:1064] 
> > Keyword DAY summary vector of different length (76 != 74)
> > ```
> 
> This is a test that I actually added, yet this should still pass....

@blattms: I quickly checked: this test (which compares the parallel to the sequential run) also fails with the flow version I had compiled from the 2024.04 release ...

File: opm/input/eclipse/Schedule/Schedule.cpp (hunk 1 above)
Comment (lisajulia) [reply opm/input/eclipse/Schedule/Schedule.cpp:1447]: No, this should not change the behavior.. hm

**Comment by blattms** [comment]
> 
> This is a test that I actually added, yet this should still pass....

I thought you added the PYACTION test [PYACTION_WELPI_INSERT_KW.DATA](https://github.com/OPM/opm-tests/pull/1190/files#diff-6128c5c0715dcc56a3190dfc4974a5987f475686282575c5755995b6ee27f9c8), but here the comparison of ACTIONX_WELPI.DATA with the reference run fails.

**Comment by lisajulia** [comment]
> > This is a test that I actually added, yet this should still pass....
> 
> I thought you added the PYACTION test [PYACTION_WELPI_INSERT_KW.DATA](https://github.com/OPM/opm-tests/pull/1190/files#diff-6128c5c0715dcc56a3190dfc4974a5987f475686282575c5755995b6ee27f9c8), but here the comparison of ACTIONX_WELPI.DATA with the reference run fails.

I've added three tests:
- one that compares PYACTION to ACTIONX (passing)
- one that compares parallel PYACTION to sequential PYACTION (failing)
- one that compares parallel ACTIONX to sequential ACTIONX (failing)

**Comment by lisajulia** [comment]
Closing, since #4136 contains everything relevant from this one

**Pull request closed without merging**